### PR TITLE
List iOS APIs requiring privacy reasons

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

As per https://developer.apple.com/documentation/bundleresources/privacy_manifest_files, 3rd party SDKs need to provide .xcprivacy files if:

- The 3rd party dependency uses one of the APIs listed in required reasons APIs. This is also relevant for 3rd party SDKs shipped as swift code = entire expo SDK.

We need to add an .xcprivacy file to expo-go to pass app review after May 1st, so we bundle xcprivacy files with expo-modules that require it.

We use this mechanism: https://github.com/SDWebImage/SDWebImage/blob/98d058a1ea053484bc4df447153654a0e4a70549/SDWebImage.podspec#L49, that I tested and confirmed to bundle correctly.

I identifed all expo-modules that make use of APIs listed in required reasons – using https://github.com/Wooder/ios_17_required_reason_api_scanner

Related to:
https://github.com/expo/expo/issues/27796
https://linear.app/expo/issue/ENG-11731/investigate-ios-privacy-manifest-requirements

# Test Plan

Tested by generating a privacy report using xCode – the items don't show since they don't add any privacy labels, but after adding the label to any of the generated xcprivacy files those labels do show up:

image

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-file-system/ios/EXFileSystemAssetLibraryHandler.m [hunk 1]
```diff
@@ -23,6 +23,7 @@ + (void)getInfoForFile:(NSURL *)fileUri
     result[@"exists"] = @(YES);
     result[@"isDirectory"] = @(NO);
     result[@"uri"] = fileUri;
+    // 3B52.1
```
Comment (aleqsio) [inline packages/expo-file-system/ios/EXFileSystemAssetLibraryHandler.m:26]:
```suggestion
    // REASON 3B52.1
```

File: packages/expo-constants/ios/EXConstantsInstallationIdProvider.m [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2015-present 650 Industries. All rights reserved.
-
+// TODO: Remove
```
Comment (aleqsio) [inline packages/expo-constants/ios/EXConstantsInstallationIdProvider.m:2]:
```suggestion
```

File: packages/expo-device/ios/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		<dict>
+			<key>NSPrivacyCollectedDataType</key>
+			<string>NSPrivacyCollectedDataTypeContacts</string>
+			<key>NSPrivacyCollectedDataTypeLinked</key>
+			<false/>
+			<key>NSPrivacyCollectedDataTypeTracking</key>
+			<false/>
+			<key>NSPrivacyCollectedDataTypePurposes</key>
+			<array>
+				<string></string>
```
Comment (lukmccall) [inline packages/expo-device/ios/PrivacyInfo.xcprivacy:16]: Can the purpose field be left empty? Will Apple reject the app for this?

File: packages/expo-device/ios/DeviceModule.swift [hunk 4]
```diff
@@ -28,6 +28,7 @@ public class DeviceModule: Module {
     }
 
     AsyncFunction("getUptimeAsync") { () -> Double in
+    // Uses required reason API based on the following reason: 35F9.1 – there's not really a matching reason here
```
Comment (lukmccall) [inline packages/expo-device/ios/DeviceModule.swift:31]:
```suggestion
      // Uses required reason API based on the following reason: 35F9.1 – there's not really a matching reason here
```

**Review by lukmccall — APPROVED** [review]

**Review by alanjhughes — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]
Looks good, thanks for taking care of that! 🙇‍♂️ 
What do you think about adding an empty xcprivacy file to the template for standalone expo modules?

File: packages/expo-device/ios/PrivacyInfo.xcprivacy (hunk 3 above)
Comment (aleqsio) [reply packages/expo-device/ios/PrivacyInfo.xcprivacy]: Good catch, this is a mistake

**Pull request merged**